geopandas/io/sql.py

```diff
@@ -324,20 +324,23 @@ def _write_postgis(
             else:
                 schema_name = "public"
 
-            target_srid = connection.execute(
-                "SELECT Find_SRID('{schema}', '{table}', '{geom_col}');".format(
-                    schema=schema_name, table=name, geom_col=geom_name
-                )
-            ).fetchone()[0]
-
-            if target_srid != srid:
-                msg = (
-                    "The CRS of the target table (EPSG:{epsg_t}) differs from the "
-                    "CRS of current GeoDataFrame (EPSG:{epsg_src}).".format(
-                        epsg_t=target_srid, epsg_src=srid
+            # Only check SRID if table exists
+            if pd.io.sql.has_table(name, connection, schema):
```

Comment:
The pandas' `has_table` implementation is basically just a call to sqlalchemy:

```
    def has_table(self, name, schema=None):
        return self.connectable.run_callable(
            self.connectable.dialect.has_table, name, schema or self.meta.schema
        )
```

so maybe we could use that directly (the `connectable` in the snippet above is the `connection` here).

Reply:
Sure, updated accordingly 👍